seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/BinaryRowType.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.api.table.type;
+
+public class BinaryRowType implements SeaTunnelDataType<BinaryRow> {
+
+    public static final BinaryRowType INSTANCE = new BinaryRowType();
+
+    @Override
+    public Class<BinaryRow> getTypeClass() {
+        return BinaryRow.class;
+    }
+
+    @Override
+    public SqlType getSqlType() {
+        return SqlType.BINARY;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        return obj instanceof PrimitiveByteArrayType;
+    }
+}
```

Comment:
When we add a new type, we have to be careful because it affects how all Sink/Transform should handle it. But the only purpose of this PR to add this new type is just to make the indicator more accurate, I think we need to discuss it in depth. cc @hailin0 @EricJoy2048